test/test_dataloader.py

```diff
@@ -1529,6 +1529,22 @@ def __len__(self):
         self.assertIsInstance(batch, torch.DoubleTensor)
         self.assertEqual(batch.size(), torch.Size([12, 2, 3, 4]))
 
+    @unittest.skipIf(not TEST_NUMPY, "numpy unavailable")
+    def test_numpy_gen_state(self):
+        from torch.utils.data._utils.worker import _generate_state
+        import numpy as np
+
+        for i in range(10):
+            # uint64 as the base seed
+            base_seed = np.random.randint(0, high=0xffffffff_ffffffff, dtype=np.uint64)
+            bs1 = np.bitwise_and(base_seed, np.uint64(0xffffffff)).item()
+            bs2 = np.right_shift(base_seed, np.uint64(32)).item()
+            for worker_id in range(8):
+                # Same seedsequence as `_generate_state`
+                ss = [worker_id, bs1, bs2, 0]
+                np_ss = np.random.SeedSequence(ss)
```

Comment:
This will mean your test suite has a dependency on numpy 1.17+, which is what you were trying to avoid by introducing `_generate_state()`. I recommend just taking the [explicit reference test vectors](https://github.com/numpy/numpy/blob/main/numpy/random/tests/test_seed_sequence.py#L7-L54) that we use in `numpy` to verify the correctness of the implementation.

Reply:
Or if you want to test this particular scheme of constructing `ss = [worker_id, ...]`, then just make your own test vectors ahead of time with `SeedSequence` and putting them explicitly here.

Reply:
Oops, thanks for pointing it out. I will generate a list of test cases for reference.